### PR TITLE
date pickers

### PR DESCRIPTION
**Pull request by nerik**
Adds full date pickers to the timeline.
screen shot 2016-08-03 at 16 55 50

Notes:
- uses https://github.com/Hacker0x01/react-datepicker, which has a de facto dependency to moment.js. Eventually we could swap all our dates to moment (except in d3), could simplify the code a bit in Timeline.jsx
- needed to add a webpack css loader to load react-datepicker css
- that `> div, > div > div, > div > div > div {` bit needed to be removed because it was fucking up react-datepicker layout. That's what happens when you style divs directly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,12 +67,14 @@
     "whatwg-fetch": "^1.0.0"
   },
   "devDependencies": {
+    "css-loader": "^0.23.1",
```
Comment (tiagojsag) [inline package.json]: will we have issues with this like we had with the svg loader? if you are not sure, move it up to prod deps, and we can revisit later.

File: app/src/components/map/DatePicker.jsx [hunk 2]
```diff
@@ -1,74 +1,48 @@
...
-  startMin: React.PropTypes.object,
-  /**
-   * Maximum allowed start date (a Date object). Depends on the inner extent start date.
-   */
-  startMax: React.PropTypes.object,
-  /**
-   * Minimum allowed end date (a Date object). Depends on the inner extent end date.
-   */
-  endMin: React.PropTypes.object,
+  minDate: React.PropTypes.object,
```
Comment (tiagojsag) [inline app/src/components/map/DatePicker.jsx]: if this is the absolute max and min dates a user can choose, they are already declared in constants.js(x)

Comment (nerik) [inline app/src/components/map/DatePicker.jsx]: Yes but those are constants representing the range of available dates for the project, while those are actually user-editable.

Comment (tiagojsag) [inline app/src/components/map/DatePicker.jsx]: :+1:

**Pull request merged**